nft/src/lib.rs

```diff
@@ -40,6 +48,8 @@ pub struct ClassInfo<TokenId, AccountId, Data> {
 	pub total_issuance: TokenId,
 	/// Class owner
 	pub owner: AccountId,
+	/// Mint mode
+	pub mint_mode: MintMode,
```

Comment:
why not just `allow_public_mint: bool`? the public API is passing `public: bool` anyway